gdrdrv/gdrdrv.c

```diff
@@ -479,16 +566,20 @@ static int gdrdrv_pin_buffer(gdr_info_t *info, void __user *_params)
         }
     }
 
-
     // here a typical driver would use the page_table to fill in some HW
     // DMA data structure
 
-    params.handle = mr->handle;
-
     mutex_lock(&info->lock);
     list_add(&mr->node, &info->mr_list);
+
+    if (gdr_generate_mr_handle(info, mr) != 0) {
+        gdr_err("No address space left for future BAR1 mapping.\n");
+        ret = -ENOMEM;
```

Comment:
same here, no point in failing when there is still address space for this mapping.